dipy/viz/horizon/app.py

```diff
@@ -451,8 +452,7 @@ def build_show(self, scene):
             for img in self.images:
                 data, affine = img
                 self.vox2ras = affine
-                img_val_range = np.unique(data).shape[0]
-                if img_val_range == 2:
+                if is_binary_image(data):
```

Comment:
> I think one of the concern was that np.unique might be too heavy on a big data. I think we should check whether if it is binary **if the roi argument** is True, not the other way around. Especially since we are repeating the same code right now (code for roi but not binary = code for not binary)

I think this logic still stands here regardless of why we need the binary image.

Reply:
@pjsjongsung If you can create an issue for me that would be amazing! so we can keep track of it.